plutus-ir/src/Language/PlutusIR/Transform/Rename.hs

```diff
@@ -53,16 +56,19 @@ Two problems arise:
    'PLC.ScopedRenameM' is for performing the renaming (the second stage).
 -}
 
-instance (PLC.HasUnique (tyname ann) PLC.TypeUnique, PLC.HasUnique (name ann) PLC.TermUnique) =>
-        PLC.Rename (Term tyname name ann) where
-    rename = PLC.runScopedRenameM . renameTermM
+type instance PLC.HasUniques (Term tyname name ann) = PLC.HasUniques (PLC.Term tyname name ann)
+
+instance PLC.HasUniques (Term tyname name ann) => PLC.Rename (Term tyname name ann) where
+    -- TODO: the Plutus Core codebase uses marking in order to prevent clashing with existing
+    -- free variables. Should we do the same here?
```

Comment:
@michaelpj ^

Reply:
Do you mean the thing where we traverse the term and make sure we've marked all the existing uniques as used?

I think in this case I just assumed that we never do anything with open PIR terms (which is true). We probably *should* handle open ones, though.

Reply:
> Do you mean the thing where we traverse the term and make sure we've marked all the existing uniques as used?

Yes. Do we have such a machinery for PIR?

Reply:
No, but we probably should. Doesn't need to be fixed here, though, leave a TODO.